### ISSUE TITLE
[Feature Request] Support for Mermaid JS (`mermaid`) in the UML Functionality

### ISSUE THREAD

**Issue by RoyiAvital** [facelessuser/MarkdownPreview#145]
Currently the [Extra module supports `UML` syntax](https://facelessuser.github.io/MarkdownPreview/extras/#uml-support) for `flow` and `sequence` by `js-sequence-diagrams` and `flowchart.js`.  
Is there a chance to add support for `mermaid` to support [`mermaid-js`](https://mermaid-js.github.io/) syntax?

Thank You.

**Comment by facelessuser** [comment]
It is possible at some point. I support Mermaid in my own personal documentation for Pymdown Extensions (something that Markdown Preview uses). But there are some limitations which you can read about here: https://facelessuser.github.io/pymdown-extensions/extras/mermaid/.

**Comment by RoyiAvital** [comment]
Yea, I enabled `mermaid` using your `pymdown-extension` but it feels it should be built in into Markdown Preview.
At least it will be great for usability.

**Comment by facelessuser** [comment]
Keep oyn mind that I added it for convenience, but I don't plan on adding everything that everyone might want. They can expand functionality by using custom fences with SuperFences and adding their own JavaScript. Something like this is low priority for me, but I will consider it.

**Comment by RoyiAvital** [comment]
Yes, I am aware of your policy for the project and understand you do it in your free time :-).

It is a request, not demand.

By the way, you should add option to donate you money.

**Comment by facelessuser** [comment]
> Yes, I am aware of your policy for the project and understand you do it in your free time :-).
>
> It is a request, not demand.

Yeah, I just like to set expectations. I keep waiting and waiting for Package Control to add support for Python 3.8 dependencies. I've been kind of waiting on that before I do some larger overhauls. I've halted backporting newer versions of Markdown and Pymdown Extensions as I keep having to backport them to Python 3.3.

> By the way, you should add option to donate you money. It won't cover the investment by yours but will show appreciation.

I do have links, just not on this specific project yet. There are a few ways to donate: sponsorship and one-time donations via PayPal. You can use the links from any project as they are not project-specific links:

https://facelessuser.github.io/pymdown-extensions/about/contributing/#become-a-sponsor

**Comment by RoyiAvital** [comment]
> I do have links, just not on this specific project yet. There are a few ways to donate: sponsorship and one-time donations via PayPal. You can use the links from any project as they are not project-specific links:
> 
> https://facelessuser.github.io/pymdown-extensions/about/contributing/#become-a-sponsor

Done! Thank You!

**Comment by facelessuser** [comment]
I decided to spend a little time this morning and add Mermaid support via PR #150.  I'm sure I have some maintenance chores to perform before I can release it as I know doc libraries and such are probably woefully out of date, but I'll start updating things so we can get this out.
